components/canvas/ProcessHierarchy/ProcessConnectionForm.tsx

```diff
@@ -20,13 +20,15 @@ type ProcessConnectionFormProps = {
   onClose: () => void;
   apiNodes: NodeDataApi[];
   apiEdges: EdgeDataApi[];
+  userCanEdit: boolean;
 };
 
 const ProcessConnectionForm = ({
   project,
   onClose,
   apiNodes,
   apiEdges,
+  userCanEdit,
```

Comment:
In order to reduce prop drilling, I would make this component check access straight from the source.

```ts
const {userCanEdit} = useAccess(project)
```